src/XrdTls/XrdTlsTempCA.cc

```diff
@@ -100,10 +100,10 @@ CASet::processFile(file_smart_ptr &fp, const std::string &fname)
     XrdCryptosslX509ParseFile(fp.get(), &chain, fname.c_str());
 
     auto ca = chain.Begin();
-    // Note we purposely leak the outputfp here; we are just borrowing the handle.
-    FILE *outputfp = fdopen(m_output_fd, "w");
-    if (!outputfp) {
+    file_smart_ptr outputfp(fdopen(dup(m_output_fd), "w"), &fclose);
```

Comment:
Can you replace the direct `dup` call with the wrapper (which sets `O_CLOEXEC`): https://github.com/xrootd/xrootd/blob/master/src/XrdSys/XrdSysFD.hh#L54

Reply:
Good idea, thank you!